### PR TITLE
pull in speedline and report speedIndex

### PR DESCRIPTION
**Pull request by samthor**
Solves #197. Requires [changes](https://github.com/pmdartus/speedline/pull/14) [in](https://github.com/pmdartus/speedline/pull/15) speedline first.

1. The speedline library is very touchy about the data it receives. This is fine, but test data now needs full frames (i.e., bigger JSON) because we're actually using them.
2. Is the goal here to drop the `first-meaningful-paint` code (all of it?) in favor of this? If so, I'm worried that Lighthouse is not structured so that e.g., I can have two results from one expensive parse operation (like speedIndex).

### CODE REVIEW

**Comment by pmdartus** [comment]
I think that both approach (`first-meaningful-paint` and `speed-index`) can live together. The main advantage of `first-meaningful-paint` is that it can be used without enabling the screenshots tracing category.

Speaking of performance, I managed increase the [frames processing speed](https://github.com/pmdartus/speedline/pull/16). But I think that we can avoid this double parsing if we can feed speedline with a DevtoolTimelineModel directly instead of a JSON. Would it solve your issue?

**Comment by brendankenny** [comment]
is it possible to reuse one of the [test traces](https://github.com/GoogleChrome/lighthouse/tree/master/test/traces/data) already in the repo/replace one of them if we need more categories than they have?

**Comment by paulirish** [comment]
Let's land this first and then sort out the traces issue next.

Sam can you sort out the test failures and get this ready to land?

File: src/aggregators/is-performant/index.js [hunk 1]
```diff
@@ -22,6 +22,9 @@ const Aggregate = require('../aggregate');
 /** @type {string} */
 const firstMeaningfulPaint = require('../../audits/performance/first-meaningful-paint').name;
 
+/** @type {string} */
+const speedIndexMetric = require('../../audits/performance/speed-index-metric').name;
```
Comment (samccone) [inline src/aggregators/is-performant/index.js]: looks like perhaps a bad path?

**Comment by samccone** [comment]
@paulirish fixed the path stuff repushed over

File: package.json [hunk 2]
```diff
@@ -37,6 +37,7 @@
     "meow": "^3.7.0",
     "npmlog": "^2.0.3",
     "semver": "^5.1.0",
+    "speedline": "^0.1.1",
```
Comment (brendankenny) [inline package.json]: 0.1.2

File: src/aggregators/is-performant/index.js (hunk 1 above)
Comment (samccone) [inline src/aggregators/is-performant/index.js]: i fixed it

File: package.json (hunk 2 above)
Comment (samccone) [inline package.json]: fixed

File: src/aggregators/is-performant/index.js (hunk 1 above)
Comment (samthor) [inline src/aggregators/is-performant/index.js]: Thanks +1

**Comment by paulirish** [comment]
> @paulirish fixed the path stuff repushed over

Nice. that's a good benefit of using branches on this repo vs personal clones. :)

**Pull request merged**